presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1847,6 +1849,211 @@ public void testDropNonExistentRole()
         assertQueryFails("DROP ROLE non_existent_role", ".*?Role 'non_existent_role' does not exist");
     }
 
+    @Test
+    public void testPublicRoleIsGrantedToAnyone()
+            throws Exception
+    {
+        Session session = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("some_user", Optional.empty()))
+                .build();
+        MaterializedResult actual = queryRunner.execute(session, "SELECT * FROM hive.information_schema.applicable_roles");
+        MaterializedResult expected = MaterializedResult.resultBuilder(session, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("some_user", "USER", "public", "NO")
+                .build();
+        assertContains(actual, expected);
+    }
+
+    @Test
+    public void testAdminRoleIsGrantedToAdmin()
+            throws Exception
+    {
+        Session session = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("admin_user", Optional.empty()))
+                .build();
+        MaterializedResult actual = queryRunner.execute(session, "SELECT * FROM hive.information_schema.applicable_roles");
+        MaterializedResult expected = MaterializedResult.resultBuilder(session, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("admin_user", "USER", "admin", "NO")
+                .build();
+        assertContains(actual, expected);
+    }
+
+    @Test
+    public void testGrantRevokeRole()
+            throws Exception
+    {
+        assertUpdate("CREATE ROLE grant_revoke_role_1");
```

Comment:
I cannot move it to the setup method. This test class may be executed concurrently.